src/modals/UpdateWalletModal.tsx

```diff
@@ -17,37 +17,96 @@ along with the library. If not, see <http://www.gnu.org/licenses/>.
 */
 
 import { BellPlus } from 'lucide-react'
+import { useEffect, useState } from 'react'
 import { useTranslation } from 'react-i18next'
 
 import InfoBox from '../components/InfoBox'
 import { Section } from '../components/PageComponents/PageContainers'
-import { openInWebBrowser } from '../utils/misc'
+import { useGlobalContext } from '../contexts/global'
+import { AlephiumWindow } from '../types/window'
 import CenteredModal, { ModalFooterButton, ModalFooterButtons } from './CenteredModal'
 
 interface UpdateWalletModalProps {
   onClose: () => void
   newVersion: string
+  startDownload?: boolean
 }
 
-const UpdateWalletModal = ({ onClose, newVersion }: UpdateWalletModalProps) => {
+type Status = 'download-available' | 'downloading' | 'download-finished' | 'download-failed'
+
+const _window = window as unknown as AlephiumWindow
+const electron = _window.electron
+
+const UpdateWalletModal = ({ onClose, newVersion, startDownload }: UpdateWalletModalProps) => {
   const { t } = useTranslation('App')
+  const { resetNewVersionDownloadTrigger } = useGlobalContext()
+
+  const [status, setStatus] = useState<Status>('download-available')
+  const [percent, setPercent] = useState(0)
+  const [error, setError] = useState('')
+
+  const newVersionDownloadTriggered = async () => {
+    setStatus('downloading')
+    electron?.updater.startUpdateDownload()
+  }
+
+  useEffect(() => {
+    const removeListener = electron?.updater.onUpdateDownloadProgress((_, info) => setPercent(info.percent))
+
+    return () => removeListener
+  }, [])
+
+  useEffect(() => {
+    const removeListener = electron?.updater.onUpdateDownloaded(() => {
+      // Delay success message to give time for download validation errors to arise if any
+      setTimeout(() => setStatus('download-finished'), 1000)
+    })
+
+    return () => removeListener
```

Comment:
Is this really cleaning the effect up? 

To do so, here we would need to call `ipcRenderer.removeListener('updater:updateDownloaded', callback)`.

Reply:
You are absolutely right. Working on it.

Reply:
You were correct, thanks for spotting this! I tested my changes and everything seems to work. Pushed.